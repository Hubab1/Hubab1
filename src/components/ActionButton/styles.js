import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
