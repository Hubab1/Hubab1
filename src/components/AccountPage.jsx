import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';
import * as Yup from 'yup';
import Grid from '@material-ui/core/Grid';
import { KeyboardDatePicker } from '@material-ui/pickers';

import { updateApplicant } from 'reducers/applicant';
import captureRoute from 'app/captureRoute';
import { H1, formContent } from 'assets/styles';
import FormTextInput from 'components/common/FormTextInput/FormTextInput';
import PhoneNumberInput from 'components/common/PhoneNumberInput';
import GenericFormError from 'components/common/GenericFormError';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { ROUTES } from 'app/constants';
import { allValuesSet } from 'utils/formik';
import { serializeDate, parseDateISOString } from 'utils/misc';	


export class AccountPage extends React.Component {
    state = {errors: null}

    get initialValues () {
        const applicant = this.props.applicant;
        let birthday = applicant.birthday;
        if (birthday) {
            birthday = parseDateISOString(birthday);
        }
        return {
            first_name: applicant.first_name,
            last_name: applicant.last_name,
            phone_number: applicant.phone_number,
            email: applicant.email,
            birthday
        }
    }

    onSubmit = (values, { setSubmitting, setErrors }) => {
        const stateUpdate = Object.assign({}, values);
        stateUpdate.birthday = serializeDate(stateUpdate.birthday);
        this.props.updateApplicant(stateUpdate, stateUpdate).then((res) => {
            if (res.errors) {
                setErrors(res.errors);
            } else {
                
            }
            setSubmitting(false);
        });
    }

    render () {
        return (
            <>
                <H1>Your Account Details</H1>
                <Formik
                    initialValues={this.initialValues}
                    validationSchema={Yup.object().shape({
                        first_name: Yup.string().required('First Name is required'),
                        last_name: Yup.string().required('Last Name is required'),
                        phone_number: Yup.string()
                            .required('Phone Number is required')
                            .matches(/^\(\d{3}\)\s\d{3}-\d{4}/, 'Must be a valid US phone number'),
                        email: Yup.string()
                            .email()
                            .required('Email is required'),
                        birthday: Yup.string()	
                            .required('required'),
                    })}
                    onSubmit={this.onSubmit}
                >
                    {({
                        values,
                        errors,
                        handleChange,
                        submitCount,
                        handleBlur,
                        handleSubmit,
                        isSubmitting,
                        touched,
                        setFieldValue,
                    }) => (
                        <form onSubmit={handleSubmit} autoComplete="off">
                            <div className={formContent}>
                                { this.state.errors && <GenericFormError errors={this.state.errors}/> }
                                <Grid container spacing={1}>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="First Name"
                                            name="first_name"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.first_name}
                                            value={values.first_name}
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="Last Name"
                                            name="last_name"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.last_name}
                                            value={values.last_name}
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="Email"
                                            name="email"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.email}
                                            value={values.email}
                                        />
                                    </Grid>
                                    <Grid item xs={6}>
                                        <PhoneNumberInput 
                                            label="Phone Number"
                                            name="phone_number"
                                            value={values.phone_number}
                                            handleChange={handleChange}
                                            error={submitCount > 0 && !!errors.phone_number}
                                            helperText={submitCount > 0 ? errors.phone_number : null}
                                        />
                                    </Grid>
                                    <Grid item xs={6}>
                                        <KeyboardDatePicker	
                                            id="birthday-picker"	
                                            clearable	
                                            disableFuture	
                                            format="MM/dd/yyyy"	
                                            placeholder="mm/dd/yyyy"	
                                            label="Birthday"	
                                            value={values.birthday || null}	
                                            fullWidth	
                                            onBlur={handleBlur}	
                                            onChange={e => setFieldValue('birthday', e)}	
                                            KeyboardButtonProps={{	
                                                'aria-label': 'change date',	
                                            }}	
                                        />	
                                    </Grid>
                                </Grid>
                                <ActionButton disabled={!allValuesSet(values) || isSubmitting} marginTop={20} marginBottom={20}>Save Changes</ActionButton>
                            </div>
                        </form>
                    )}
                </Formik>
            </>
        );
    }
}

AccountPage.propTypes = {
    updateApplicant: PropTypes.func,
}

const mapStateToProps = (state) => ({
    applicant: state.applicant,
});

const mapDispatchToProps = { updateApplicant };

export default connect(mapStateToProps, mapDispatchToProps)(captureRoute(AccountPage, ROUTES.ACCOUNT));
