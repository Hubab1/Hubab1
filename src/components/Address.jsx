import React, { useState, useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { TextField } from 'formik-material-ui';
import { Formik, Form, Field } from 'formik';
import * as Yup from 'yup';
import styled from '@emotion/styled';

import { ROUTES } from 'app/constants';
import { updateApplicant } from 'reducers/applicant';
import withRelativeRoutes from 'app/withRelativeRoutes';
import ActionButton from 'components/common/ActionButton/ActionButton';
import LocationSearch from 'components/common/Formik/LocationSearch/LocationSearch';
import { H1, SpacedH3 } from 'assets/styles';
import sticky from 'assets/images/sticky.png';
import GenericFormMessage from './common/GenericFormMessage';

const ImageContainer = styled.div`
    margin-top: 31px;
    margin-bottom: 31px;
    img {
        max-height: 105px;
        max-width: 114px;
    }
`;

export const validationSchema = Yup.object().shape({
    address_search: Yup.object().shape({
        address_street: Yup.string()
            .required('Street is required')
            .matches(/^[A-Za-z0-9]+(?:\s[A-Za-z0-9'_-]+)+$/, 'Invalid street'),
        address_city: Yup.string()
            .required('City is required')
            .matches(/^[\w'\-,.][^0-9_!¡?÷?¿/\\+=@#$%ˆ&*(){}|~<>;:[\]]{1,}$/, 'Invalid city'),
        address_state: Yup.string().required('State is required'),
        address_postal_code: Yup.number().required('Zip code is required'),
    }),
    address_line_2: Yup.string(),
});

export const Address = ({ applicant, updateApplicant, _nextRoute }) => {
    const [errors, setErrors] = useState(null);
    const handleSubmit = useCallback(
        async (values, { setSubmitting, setErrors: setFormErrors }) => {
            try {
                const serialized = {
                    ...values.address_search,
                    address_line_2: values.address_line_2,
                };

                const response = await updateApplicant(serialized);
                if (response.errors) {
                    setFormErrors(response.errors);
                } else {
                    _nextRoute();
                }
            } catch {
                setErrors(['Oops! We ran into some issues. Please try again later.']);
            } finally {
                setSubmitting(false);
            }
        },
        [updateApplicant, _nextRoute]
    );

    const initialValues = useMemo(() => {
        const searchBuilder = [];
        if (applicant.address_street) searchBuilder.push(applicant.address_street);
        if (applicant.address_city) searchBuilder.push(applicant.address_city);
        if (applicant.address_state) searchBuilder.push(applicant.address_state);
        if (applicant.address_postal_code) searchBuilder.push(applicant.address_postal_code);
        const search = searchBuilder.join(', ');

        return {
            address_search: {
                search,
                address_street: applicant.address_street,
                address_city: applicant.address_city,
                address_state: applicant.address_state,
                address_postal_code: applicant.address_postal_code,
            },
            address_line_2: applicant.address_line_2,
        };
    }, [applicant]);

    if (!applicant) {
        return null;
    }

    return (
        <>
            <H1>Tell Us A Little More</H1>
            <SpacedH3>Now, by filling out these details below we can screen you more accurately.</SpacedH3>
            <ImageContainer>
                <img src={sticky} alt="sticky note" />
            </ImageContainer>
            <Formik validationSchema={validationSchema} initialValues={initialValues} onSubmit={handleSubmit}>
                {({ values, isSubmitting, handleSubmit, submitCount }) => {
                    const disableSubmit = !values.address_search.search || isSubmitting;

                    return (
                        <Form onSubmit={handleSubmit} autoComplete="off">
                            <Field
                                fullWidth
                                margin="normal"
                                label="Street name, city, state, zip"
                                name="address_search"
                                submitCount={submitCount}
                                component={LocationSearch}
                            />
                            <Field
                                fullWidth
                                margin="normal"
                                label="Apt/Ste/Floor"
                                name="address_line_2"
                                component={TextField}
                            />
                            <GenericFormMessage type="error" messages={errors} />
                            <ActionButton marginTop={50} disabled={disableSubmit}>
                                Continue
                            </ActionButton>
                        </Form>
                    );
                }}
            </Formik>
        </>
    );
};

Address.propTypes = {
    updateApplicant: PropTypes.func.isRequired,
    applicant: PropTypes.object,
    _nextRoute: PropTypes.func,
};

const mapStateToProps = (state) => ({
    applicant: state.applicant,
});

const mapActionsToProps = {
    updateApplicant,
};

export default connect(mapStateToProps, mapActionsToProps)(withRelativeRoutes(Address, ROUTES.ADDRESS));
