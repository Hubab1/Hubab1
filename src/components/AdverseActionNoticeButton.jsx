import React, { useState } from 'react';
import API from 'app/api';
import { LinkButton } from 'assets/styles';
import { css } from 'emotion';

export const aanLink = css`
    outline: none;
`;

export function AdverseActionNoticeButton() {
    const [error, setError] = useState(false);

    const openDocument = async () => {
        const w = window.open('data:text/plain;charset=utf-8;base64,TG9hZGluZw==', '_blank');
        try {
            const response = await API.fetchAANDocument();
            const blobUrl = URL.createObjectURL(new Blob([response], { type: 'application/pdf' }));
            w.location.replace(blobUrl);
        } catch (e) {
            setError(true);
        }
    };

    return (
        <LinkButton onClick={openDocument} className={aanLink}>
            {error ? 'An error occurred. Please try again.' : 'Learn why'}
        </LinkButton>
    );
}
