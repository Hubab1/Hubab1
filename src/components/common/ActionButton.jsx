import React from 'react';

import Button from '@material-ui/core/Button';
import { root, label } from 'assets/emotion/ActionButton';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children } = this.props;
        return (
            <Button
                onClick={onClick}
                classes={{ root, label }}
                variant="contained"
                color={color || 'primary'}
                type="submit"
                disabled={disabled}
                fullWidth
            >
                {children}
            </Button>
        );
    }
}