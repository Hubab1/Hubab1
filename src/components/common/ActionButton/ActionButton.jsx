import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import { root, label, ButtonContainer } from './styles';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children, marginTop, marginBottom, variant } = this.props;
        return (
            <ButtonContainer marginTop={marginTop} marginBottom={marginBottom}>
                <Button
                    onClick={onClick}
                    classes={{ root, label }}
                    variant={variant}
                    color={color}
                    type="submit"
                    disabled={disabled}
                    fullWidth
                >
                    {children}
                </Button>
            </ButtonContainer>
        );
    }
}

ActionButton.propTypes = {
    color: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    marginTop: PropTypes.string,
    marginBottom: PropTypes.string,
    variant: PropTypes.string,
}

ActionButton.defaultProps = {
    variant: 'contained',
    color: 'primary',
}