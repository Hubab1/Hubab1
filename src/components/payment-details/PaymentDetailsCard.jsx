import { styles } from 'components/payment-details/styles';
import styled from '@emotion/styled';
import lightbulb from 'assets/images/lightbulb.png';
import { PaymentDetailRows } from 'components/payment-details/PaymentDetailRow';
import React from 'react';
import { format, parseISO } from 'date-fns';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import PaidText from 'components/common/PaidText';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Card } from 'assets/styles';

const useStyles = makeStyles(() => ({
    root: {
        display: 'block',
        padding: 0,
    },
}));

export const PaymentCard = styled(Card)`
    padding: 15px;
`;

export const PaymentDetailsCard = ({ profile, payables = [] }) => {
    const leaseStartDate = format(parseISO(profile.lease_start_date), 'MM/dd/yyyy');
    const allPaymentsPaid = !payables.find((payment) => !payment.paid);

    return (
        <PaymentCard>
            <div className={styles.headerContainer}>
                <div className={styles.imageWrapper}>
                    <img alt="coin" src={lightbulb} />
                </div>
                <div className={styles.cardHeader}>
                    <div className={styles.title}>Payment breakdown</div>
                    <div>Move in date: {leaseStartDate}</div>
                    <div>Lease term: {profile.lease_term} months</div>
                </div>
            </div>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.application_fees.total}
                label={'due at application'}
                defaultExpanded={false}
                isPaid={allPaymentsPaid}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.application_fees} paymentType="application" />
            </PaymentItemsExpansionPanel>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.move_in_fees_v2.total}
                label={'due at move in'}
                defaultExpanded={false}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.move_in_fees_v2} paymentType="move_in" />
            </PaymentItemsExpansionPanel>
            <PaymentItemsExpansionPanel
                amount={profile.fees_breakdown.monthly_fees.total}
                label={'monthly rent'}
                defaultExpanded={false}
            >
                <PaymentDetailRows paymentObject={profile.fees_breakdown.monthly_fees} paymentType="monthly_fees" />
            </PaymentItemsExpansionPanel>
        </PaymentCard>
    );
};

PaymentDetailsCard.propTypes = {
    profile: PropTypes.object.isRequired,
    payables: PropTypes.array,
};

export function PaymentItemsExpansionPanel({ children, defaultExpanded, label, amount, isPaid }) {
    const classes = useStyles();

    return (
        <div className={styles.itemsContainer}>
            <ExpansionPanel elevation={0} defaultExpanded={defaultExpanded}>
                <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
                    <div className={styles.itemsContainerLabel}>
                        <b>${amount}</b>&nbsp;{label}
                    </div>
                    {isPaid && (
                        <div className={styles.paidText}>
                            <PaidText />
                        </div>
                    )}
                </ExpansionPanelSummary>
                <ExpansionPanelDetails classes={{ root: classes.root }}>
                    {React.Children.map(children, (child) => (
                        <div className={styles.itemRow}>{child}</div>
                    ))}
                </ExpansionPanelDetails>
            </ExpansionPanel>
        </div>
    );
}

PaymentItemsExpansionPanel.propTypes = {
    children: PropTypes.object,
    defaultExpanded: PropTypes.bool,
    label: PropTypes.string.isRequired,
    amount: PropTypes.string.isRequired,
    isPaid: PropTypes.bool,
};
