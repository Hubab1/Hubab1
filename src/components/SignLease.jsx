import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { applicantUpdated } from 'reducers/applicant';
import API from 'app/api';
import hsclient from 'utils/hsclient';
import { ROUTES, HELLOSIGN_TEST_MODE, APPLICATION_EVENTS, DOCUMENT_TYPE_LEASE } from 'app/constants';
import { H1, SpacedH3, blackLinkRoot, arrowIcon } from 'assets/styles';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { useEffect } from 'react';
import captureRoute from 'app/captureRoute';
import { fetchPayments } from 'reducers/payments';
import { PaymentDetailsCard } from 'components/payment-details/PaymentDetailsCard';
import { Link } from 'react-router-dom';
import ArrowBackIos from '@material-ui/icons/ArrowBackIos';

export const SignLease = ({ profile, configuration, history, applicantUpdated, payables }) => {
    useEffect(() => {
        fetchPayments();

        hsclient.on('sign', async () => {
            // ensure FE has the applicant signed milestone before navigating to next screen
            const newApplicant = await API.fetchApplicant();
            const leaseSignedMilestone = newApplicant.events.find(
                (e) => parseInt(e.event) === parseInt(APPLICATION_EVENTS.MILESTONE_APPLICANT_SIGNED_LEASE)
            );
            if (!leaseSignedMilestone) {
                newApplicant.events.push({
                    event: APPLICATION_EVENTS.MILESTONE_APPLICANT_SIGNED_LEASE,
                    milestone: true,
                });
            }
            applicantUpdated(newApplicant);
            // lease may not be ready by the time of navigation to the lease signed page
            setTimeout(() => history.push(ROUTES.LEASE_SIGNED), 2500);
        });
        return () => {
            hsclient.off('sign');
        };
    }, [applicantUpdated, history]);

    if (!profile || !configuration) return null;

    const openEmbeddedSigning = async () => {
        const data = await API.embeddedSigningUrl(DOCUMENT_TYPE_LEASE);
        if (data.url) {
            hsclient.open(data.url, {
                testMode: HELLOSIGN_TEST_MODE,
                skipDomainVerification: HELLOSIGN_TEST_MODE,
                allowCancel: false,
                allowDecline: false,
            });
        }
    };

    return (
        <>
            <div>
                <H1>{`Payment Details`}</H1>
                <SpacedH3>{`Make sure everything looks good before signing the lease.`}</SpacedH3>
                <PaymentDetailsCard profile={profile} payables={payables} />
            </div>
            <ActionButton onClick={openEmbeddedSigning} marginTop={30}>
                View & Sign Lease
            </ActionButton>
            <br />
            <Link to={ROUTES.APP_APPROVED} className={blackLinkRoot}>
                <ArrowBackIos classes={{ root: arrowIcon }} /> Go Back
            </Link>
        </>
    );
};

SignLease.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
    updateApplicant: PropTypes.object,
    applicant: PropTypes.object,
    history: PropTypes.object,
    applicantUpdated: PropTypes.func,
    payables: PropTypes.array,
};

const mapStateToProps = (state) => ({
    profile: state.renterProfile,
    applicant: state.applicant,
    configuration: state.configuration,
    payables: state.payments || [],
});

const mapDispatchToProps = {
    applicantUpdated,
    fetchPayments,
};

export default connect(mapStateToProps, mapDispatchToProps)(captureRoute(SignLease, ROUTES.SIGN_LEASE));
