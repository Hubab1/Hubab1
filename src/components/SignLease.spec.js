import React from 'react';
import { shallow } from 'enzyme';
import ActionButton from 'components/common/ActionButton/ActionButton';
import API from 'app/api';
import { SignLease } from 'components/SignLease';
import { PaymentDetailsCard } from 'components/payment-details/PaymentDetailsCard';
import { Link } from 'react-router-dom';
import { ROUTES } from 'app/constants';

var mockhsclient;
jest.mock('utils/hsclient', () => {
    class HSclient {
        events = {};
        open = () => {};
        on = (name, cb) => {
            this.events[name] = cb;
        };
        _trigger = (name) => {
            this.events[name]();
        };
    }
    mockhsclient = new HSclient();
    return mockhsclient;
});

const buildProps = (buildingName = 'Fake Building', streetAddress = '123 Fake Street', unitNumber = '2B') => {
    return {
        profile: {
            events: [{ event: '270' }],
            unit: {
                unit_number: unitNumber,
            },
            last_status_change: {
                created_at: '2020-05-06 04:06:23',
            },
        },
        configuration: {
            community: {
                building_name: buildingName,
                normalized_street_address: streetAddress,
            },
        },
        history: {
            push: jest.fn(),
        },
        applicant: {
            client: {
                person: {
                    name: 'John Doe',
                },
            },
        },
        applicantUpdated: jest.fn(),
        fetchPayments: jest.fn(),
        payables: [],
    };
};

describe('renders the component', () => {
    const props = buildProps('Fake Building', '123 Fake Street', null);
    const wrapper = shallow(<SignLease {...props} />);

    it('matches snapshot', () => {
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders the correct elements', () => {
        expect(wrapper.find(PaymentDetailsCard)).toHaveLength(1);
        expect(wrapper.find(ActionButton)).toHaveLength(1);
        expect(wrapper.find(Link).prop('to')).toBe(ROUTES.APP_APPROVED);
    });
});

describe('hellosign modal', () => {
    it('fetches embedded signing url before opening', () => {
        const props = buildProps('Fake Building', '123 Fake Street', null);
        API.embeddedSigningUrl = jest.fn().mockReturnValue({ url: 'test' });
        API.fetchApplicant = jest.fn().mockReturnValue({ events: [] });
        const wrapper = shallow(<SignLease {...props} />);
        wrapper.find(ActionButton).simulate('click');
        expect(API.embeddedSigningUrl).toHaveBeenCalled();
    });
});
