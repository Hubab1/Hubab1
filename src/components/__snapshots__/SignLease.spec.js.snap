// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component matches snapshot 1`] = `
<React.Fragment>
  <div>
    <ForwardRef(render)>
      Payment Details
    </ForwardRef(render)>
    <ForwardRef(render)>
      Make sure everything looks good before signing the lease.
    </ForwardRef(render)>
    <PaymentDetailsCard
      payables={Array []}
      profile={
        Object {
          "events": Array [
            Object {
              "event": "270",
            },
          ],
          "last_status_change": Object {
            "created_at": "2020-05-06 04:06:23",
          },
          "unit": Object {
            "unit_number": null,
          },
        }
      }
    />
  </div>
  <ActionButton
    color="primary"
    marginTop={30}
    onClick={[Function]}
    successGreen={false}
    variant="contained"
  >
    View & Sign Lease
  </ActionButton>
  <br />
  <ForwardRef
    className="css-vq6rzm"
    to="/application-approved"
  >
    <UNDEFINED
      classes={
        Object {
          "root": "css-13z8uwt",
        }
      }
    />
     Go Back
  </ForwardRef>
</React.Fragment>
`;
